pkg/utils/certificateUtils.go

```
@@ -17,33 +17,106 @@ package utils
+				return nil
 			}
-			if cert != nil {
-				certs = append(certs, cert...)
+
+			targetFileInfo, err = os.Lstat(targetFilePath)
+
+			if err != nil {
+				logrus.Errorf("error getting file info for path '%v', error '%v'", targetFilePath, err)
+				return nil
 			}
 		}
+
+		if targetFileInfo != nil && !targetFileInfo.IsDir() && !isSymbolicLink(targetFileInfo) {
+			if _, ok := fileMap[targetFilePath]; !ok {
+				certs, err = loadCertFile(targetFileInfo, targetFilePath, certs, fileMap)
+				if err != nil {
+					return errors.Wrap(err, "error reading certificate file "+targetFilePath)
+				}
+			}
+		}
+
 		return nil
 	})
 
 	if err != nil {
 		return nil, err
 	}
 
+	logrus.Infof("%v notary verification certificates loaded from path '%v'", len(certs), path)
 	return certs, nil
 }
+
+func isSymbolicLink(info fs.FileInfo) bool {
+	return info.Mode()&os.ModeSymlink != 0
+}
+
+func loadCertFile(fileInfo fs.FileInfo, filePath string, certificate []*x509.Certificate, fileMap map[string]bool) ([]*x509.Certificate, error) {
+
+	result := certificate
+	cert, certError := cryptoutil.ReadCertificateFile(filePath) // ReadCertificateFile returns empty if file was not a certificate
+	if certError != nil {
+		return certificate, certError
+	}
+	if cert != nil {
+		result = append(certificate, cert...)
+		fileMap[filePath] = true
+	}
+
+	return result, nil
+}
+
+// Replace the shortcut prefix in a path with the home directory
+// For example in a unix os, ~/.config/ becomes /home/azureuser/.config after replacement
+func ReplaceHomeShortcut(path string) string {
+
+	shortcutPrefix := homedir.GetShortcutString() + string(os.PathSeparator)
+	if strings.HasPrefix(path, shortcutPrefix) {
+		home := homedir.Get()
+		if len(home) > 0 {
+			return strings.Replace(path, homedir.GetShortcutString(), home, 1) // replace 1 instance
+		} else {
+			logrus.Warningf("Path '%v' replacement failed , value of Home dir '%v'", path, home)
```

Should we return an error for this function or is it ok to continue if the home path replacement fails?

---

my preference would be to be non blocking , and let it fail on actual cert load,  :)